opencti-platform/opencti-front/src/private/components/common/stix_nested_ref_relationships/StixNestedRefRelationshipCreationFromKnowledgeGraph.tsx

```diff
@@ -0,0 +1,67 @@
+import { stixNestedRefRelationshipCreationResolveQuery } from '@components/common/stix_nested_ref_relationships/StixNestedRefRelationshipCreation';
+import IconButton from '@mui/material/IconButton';
+import { ReadMoreOutlined } from '@mui/icons-material';
+import Tooltip from '@mui/material/Tooltip';
+import React, { FunctionComponent } from 'react';
+import {
+  StixNestedRefRelationshipCreationResolveQuery$data,
+} from '@components/common/stix_nested_ref_relationships/__generated__/StixNestedRefRelationshipCreationResolveQuery.graphql';
+import { NodeObject } from 'react-force-graph-2d';
+import { QueryRenderer } from '../../../../relay/environment';
+import { useFormatter } from '../../../../components/i18n';
+
+interface StixNestedRefRelationshipCreationFromKnowledgeGraphProps {
+  nestedRelationExist: boolean,
+  openCreateNested: boolean,
+  nestedEnabled: boolean,
+  relationFromObjects: NodeObject[],
+  relationToObjects: NodeObject[],
+  handleSetNestedRelationExist: (val: boolean) => void,
+  handleOpenCreateNested: () => void,
+}
+const StixNestedRefRelationshipCreationFromKnowledgeGraph: FunctionComponent<StixNestedRefRelationshipCreationFromKnowledgeGraphProps> = ({
+  nestedRelationExist,
+  openCreateNested,
+  nestedEnabled,
+  relationFromObjects,
+  relationToObjects,
+  handleSetNestedRelationExist,
+  handleOpenCreateNested,
+}) => {
+  const { t_i18n } = useFormatter();
+  return (
+    <Tooltip title={t_i18n('Create a nested relationship')}>
+      <>
+        {(nestedEnabled && relationFromObjects[0] && relationToObjects[0] && !openCreateNested) && (
+        <QueryRenderer
```

Comment:
If you have the time, could you remove this old `QueryRenderer` to use hooks instead?

Reply:
done :)